src/main/scala/org/apache/spark/sql/delta/hooks/GenerateManifest.scala

```diff
@@ -214,58 +221,46 @@ trait GenerateSymlinkManifestImpl extends PostCommitHook with DeltaLogging with
    * @param hadoopConf           Hadoop configuration to use
    * @return Set of partition relative paths of the written manifest files (e.g., part1=1/part2=2)
    */
-  private def writeManifestFiles(
-      deltaLogDataPath: Path,
-      manifestRootDirPath: String,
-      fileNamesForManifest: Dataset[AddFile],
-      partitionCols: Seq[String],
-      hadoopConf: SerializableConfiguration): Set[String] = {
-
+  protected def writeManifestFiles(
+       deltaLogDataPath: Path,
+       manifestRootDirPath: String,
+       fileNamesForManifest: Dataset[AddFile],
+       partitionCols: Seq[String],
+       hadoopConf: SerializableConfiguration): Set[String] = {
     val spark = fileNamesForManifest.sparkSession
     import spark.implicits._
 
     val tableAbsPathForManifest =
       LogStore(spark.sparkContext).resolvePathOnPhysicalStorage(deltaLogDataPath).toString
 
-    /** Write the data file relative paths to manifestDirAbsPath/manifest as absolute paths */
-    def writeSingleManifestFile(
```

Comment:
Ideally this is the only function that should change with both the implementations

Reply:
yeah, but I set different path locations for JSON & Symlink,  and introduced ManifestType to cover the detail differences between the JSON and Symlink.
Feel free to edit the codes if you have a better idea ~.